app.go

```diff
@@ -81,9 +83,21 @@ func (f optionFunc) apply(app *App) { f(app) }
 //
 // See the documentation for go.uber.org/dig for further details.
 func Provide(constructors ...interface{}) Option {
-	return optionFunc(func(app *App) {
-		app.provides = append(app.provides, constructors...)
-	})
+	return provideOption(constructors)
+}
+
+type provideOption []interface{}
+
+func (po provideOption) apply(app *App) {
+	app.provides = append(app.provides, po...)
```

Comment:
That change will require returning an error from `New`, which I think is
undesirable.

Instead of a minimal app being,

```go
app := fx.New()
app.Run()
```

It will become,

```go
app, err := fx.New()
if err != nil {
  log.Fatal(err)
}

app.Run()
```